presto-main/src/test/java/com/facebook/presto/util/TestGraphvizPrinter.java

```diff
@@ -0,0 +1,133 @@
+import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.plan.PlanNodeId;
+import com.facebook.presto.spi.plan.TableScanNode;
+import com.facebook.presto.sql.planner.Partitioning;
+import com.facebook.presto.sql.planner.PartitioningScheme;
+import com.facebook.presto.sql.planner.PlanFragment;
+import com.facebook.presto.sql.planner.SubPlan;
+import com.facebook.presto.sql.planner.plan.PlanFragmentId;
+import com.facebook.presto.testing.TestingTransactionHandle;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.FunctionAndTypeManager.createTestFunctionAndTypeManager;
+import static com.facebook.presto.sql.planner.SystemPartitioningHandle.SINGLE_DISTRIBUTION;
+import static com.facebook.presto.sql.planner.SystemPartitioningHandle.SOURCE_DISTRIBUTION;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+
+public class TestGraphvizPrinter
+{
+    private static final String TEST_CONNECTOR_ID_STR = "connector_id";
+    private static final ConnectorId TEST_CONNECTOR_ID = new ConnectorId(TEST_CONNECTOR_ID_STR);
+    private static final PlanNodeId TEST_TABLE_SCAN_NODE_ID = new PlanNodeId("plan_id");
+    private static final String TEST_CONNECTOR_TABLE_HANDLE_STR = "testConnectorTableHandle";
+    private static final ConnectorTableHandle TEST_CONNECTOR_TABLE_HANDLE = new ConnectorTableHandle()
+    {
+        @Override
+        public String toString()
+        {   // TestingMetadata::TestingTableHandle uses default toString, which is inconsistent
+            return TEST_CONNECTOR_TABLE_HANDLE_STR;
+        }
+    };
+    private static final PlanNode TEST_TABLE_SCAN_NODE = new TableScanNode(
+            TEST_TABLE_SCAN_NODE_ID,
+            new TableHandle(TEST_CONNECTOR_ID, TEST_CONNECTOR_TABLE_HANDLE, TestingTransactionHandle.create(), Optional.empty()),
+            ImmutableList.of(),
+            ImmutableMap.of(),
+            TupleDomain.all(),
+            TupleDomain.all());
+    private static final String TEST_TABLE_SCAN_NODE_ATTR_STR = format(
+            "label=\"{TableScan[TableHandle \\{connectorId='%s', connectorHandle='%s', layout='Optional.empty'\\}]}\", style=\"rounded, filled\", shape=record, fillcolor=deepskyblue",
+            TEST_CONNECTOR_ID_STR,
+            TEST_CONNECTOR_TABLE_HANDLE_STR);
+
+    @Test
+    public void testPrintLogical()
+    {
+        String actualStr = GraphvizPrinter.printLogical(
+                ImmutableList.of(createTestPlanFragment(0, TEST_TABLE_SCAN_NODE)),
+                testSessionBuilder().build(),
+                createTestFunctionAndTypeManager());
+        String expectStr = String.join(
+                "\n",
```

Comment:
use %n instead of \n, because \n is not portable across different systems. Or use System.lineSeparator()

Reply:
done (used `System.lineSeparator()`)